File: app/src/main/java/net/oneread/oneread/util/JsonUtil.java
package net.oneread.oneread.util;

import com.google.gson.Gson;
import com.google.gson.JsonElement;
import com.google.gson.JsonSyntaxException;

/**
 * Created by sermilion on 9/12/17.
 */

public class JsonUtil {
    private static Gson gson;

    public static Gson getDefaultGson() {
        if (gson == null) {
            gson = new Gson();
        }
        return gson;
    }

    public static String toJson(Object object) {
        return getDefaultGson().toJson(object);
    }

    public static <T> T fromJson(String json, Class<T> classOfT) throws JsonSyntaxException {
        return getDefaultGson().fromJson(json, classOfT);
    }

    public static String childToString(JsonElement element, String childName) {
        if (element == null || element.isJsonNull()) {
            return null;
        }
        final JsonElement child = element.getAsJsonObject().get(childName);
        if (child == null || child.isJsonNull()) {
            return null;
        }
        return child.getAsString();
    }
}

File: app/src/main/java/net/oneread/oneread/util/JsonUtil.kt
package net.oneread.oneread.util

import com.google.gson.Gson
import com.google.gson.JsonElement
import com.google.gson.JsonSyntaxException

/**
 * Created by sermilion on 9/12/17.
 */

object JsonUtil {
    private lateinit var gson: Gson

    val defaultGson: Gson
        get() {
            if (gson == null) {
                gson = Gson()
            }
            return gson
        }

    fun toJson(`object`: Any): String = defaultGson.toJson(`object`)

    @Throws(JsonSyntaxException::class)
    fun <T> fromJson(json: String, classOfT: Class<T>): T = defaultGson.fromJson(json, classOfT)

    fun childToString(element: JsonElement?, childName: String): String? {
        if (element == null || element.isJsonNull) {
            return null
        }
        val child = element.asJsonObject.get(childName)
        return if (child == null || child.isJsonNull) {
            null
        } else child.asString
    }
}